All the test suites are failing

### Package

<!-- Which package is this for? (Delete as appropriate) -->

`graphql-hooks`
`graphql-hooks-ssr`
`graphql-hooks-memcache`

### Description

When running `npm run test`, from the root of the project, all the unit test fail with a similar message (these are just a few examples):
```
error TS2339: Property 'toBe' does not exist on type 'Assertion'.
error TS2339: Property 'toHaveBeenCalled' does not exist on type 'Assertion'.
error TS2339: Property 'rejects' does not exist on type 'Assertion'.
```
### How to reproduce

Run `npm run test`.

### Suggested solution (optional)

Most likely the reason for the failures is a misconfiguration/or missing some config. 

Part of fixing this issue should be also to remove the `tsd` dependency and type tests (located in `/packages/graphql-hooks/test-d/` folder) as they are **most likely not needed anymore** since we are using `tsc` to check the types (check this [comment](https://github.com/nearform/graphql-hooks/pull/799#discussion_r833635348) and [PR](https://github.com/nearform/graphql-hooks/pull/799) for more information).

--- a/babel.config.js
+++ b/babel.config.js
@@ -9,6 +9,7 @@ module.exports = {
         }
       }
     ],
-    '@babel/preset-react'
+    '@babel/preset-react',
+    '@babel/preset-typescript'
   ]
 }

--- a/jest.config.js
+++ b/jest.config.js
@@ -1,11 +1,5 @@
 const commonConfig = {
-  preset: 'ts-jest/presets/js-with-ts',
-  testEnvironment: 'jsdom',
-  globals: {
-    'ts-jest': {
-      tsconfig: '<rootDir>/config/tsconfig.base.json'
-    }
-  }
+  testEnvironment: 'jsdom'
 }
 
 const projects = [
@@ -47,7 +41,7 @@ const projects = [
       roots: ['./examples/create-react-app'],
       coveragePathIgnorePatterns: ['/node_modules/', '<rootDir>/packages/*'],
       transform: {
-        '\\.jsx?$': 'babel-jest'
+        '\\.[jt]sx?$': 'babel-jest'
       },
       displayName: 'cra-example',
       testEnvironment: 'jsdom'

--- a/package.json
+++ b/package.json
@@ -22,6 +22,7 @@
     "@babel/eslint-parser": "^7.16.5",
     "@babel/preset-env": "^7.14.1",
     "@babel/preset-react": "^7.13.13",
+    "@babel/preset-typescript": "^7.18.6",
     "@commitlint/cli": "^17.0.3",
     "@commitlint/config-conventional": "^17.0.3",
     "@testing-library/cypress": "^8.0.3",
@@ -50,7 +51,7 @@
     "rollup-plugin-replace": "^2.2.0",
     "rollup-plugin-size-snapshot": "^0.12.0",
     "rollup-plugin-terser": "^7.0.0",
-    "ts-jest": "^29.0.0-next.0",
+    "ts-jest": "^29.0.1",
     "typescript": "^4.7.4"
   },
   "lint-staged": {

--- a/packages/graphql-hooks-ssr/index.test.tsx
+++ b/packages/graphql-hooks-ssr/index.test.tsx
@@ -1,3 +1,4 @@
+import React from 'react'
 import { getInitialState } from './index'
 
 describe('getInitialState', () => {

